source/Irrlicht/CGLTFMeshFileLoader.cpp

```diff
@@ -227,24 +239,49 @@ IAnimatedMesh* CGLTFMeshFileLoader::createMesh(io::IReadFile* file)
 		return nullptr;
 	}
 
-	const auto indicesAccessorId =
-		model.meshes[0].primitives[0].indices;
-	const auto positionAccessorId =
-		model.meshes[0].primitives[0].attributes["POSITION"];
-
-	auto* indices = getIndices(model, indicesAccessorId);
-	auto* vertices = getVertices(model, positionAccessorId);
-
-	SMeshBuffer* meshbuf { new SMeshBuffer {} };
+	// Create the base mesh
+	SMesh* mesh { new SMesh {} };
 
-	meshbuf->append(vertices, model.accessors[positionAccessorId].count,
-		indices, model.accessors[indicesAccessorId].count);
+	// Iterate models
+	for (std::size_t mesh_index = 0;
+		mesh_index < model.meshes.size(); mesh_index++) {
+	// Iterate primitives
+	for (std::size_t primitive_index = 0; primitive_index < model
+		.meshes[mesh_index].primitives.size(); primitive_index++) {
+
+		const auto positionAccessorId = model.meshes[mesh_index]
+			.primitives[primitive_index].attributes["POSITION"];
+		const auto indicesAccessorId = model.meshes[mesh_index]
+			.primitives[primitive_index].indices;
+
+		// Creates counts for preallocation
+		std::size_t vertices_count = model.accessors[positionAccessorId].count;
+
+		// We must count to create containers for the data	
+		// Create new buffer for vertices, positions, and normals
+		auto* vertexBuffer = new video::S3DVertex[vertices_count]{};
+		// This is used to copy data into the vertexBuffer
+		Span<video::S3DVertex> verticesBuffer{ vertexBuffer, vertices_count };
+		// Create dynamic indices buffer so it's easier to work with
+		std::vector<u16> indicesBuffer;

```

Comment:
We know the number of indices, so might as well do `indicesBuffer(count)` to preallocate space in the vector.